_includes/translated-warning.html

```diff
@@ -1,5 +1,11 @@
+{% assign urlpathparts = page.url | split: "/" %}
+{% assign currlangcode = urlpathparts[1] %}
+{% assign officialpage = page.url %}
+{% assign prefixtoremove = "/" | append: currlangcode %}
+{% assign officialpage = officialpage | remove_first: prefixtoremove %}
```

Comment:
This assumes URL of the original page to be the same (minus lang prefix) than the original page. This is currently the case, but if we ever decide otherwise, this will break. The language switcher uses different method to find different-lang page URL: https://github.com/marmarek/qubesos.github.io/blob/translation-output/_includes/header.html#L32
Better use the same method here (but look specifically for original language version, not all of them.

Reply:
> This assumes URL of the original page to be the same (minus lang prefix) than the original page.

This was my original idea since it is easy (and quick!) to calculate. I'm a little afraid of that the other method could lead to performance issues.

Reply:
Performance is not an issue at all. Jekyll generates static html files, so at the worst case, the website generation (one time after changing something) will be slower.

Reply:
> […] at the worst case, the website generation (one time after changing something) will be slower.

This is what I mean. ;-)  We will see …

Reply:
Using the filter pipeline

```
{% assign posts = site.pages | concat: site.doc | concat: site.qubes-translated | where:'ref', page.ref | sort: 'lang' %}
```

(or something like that) costs a lot perfomance. The performance seems to be dominated by both using the filter and the number of files to render. It's around O(number_of_times_using_the_filter * number_of_Markdown_files). I used my watch and tested roughly.

Reply:
It seems that this
```
{% assign posts = site.pages | concat: site.doc | concat: site.qubes-translated %}
```
consumes the most time. Unfortunately, it seems that something like this
```
{% assign posts = site.pages | concat: site.doc | concat: site.qubes-translated %}
{% assign x = posts | where: 'ref', page.ref %}
...
{% assign x = posts | where: 'ref', page.ref %}
...
```
doesn't seem to work all times. Also copying like
```
{% assign postsc = site.pages | concat: site.doc | concat: site.qubes-translated %}
{% assign posts = postsc %}
{% assign x = posts | where: 'ref', page.ref %}
...
{% assign posts = postsc %}
{% assign x = posts | where: 'ref', page.ref %}
...
```
doesn't work. So, we probably have to write the concats every time. :(